### DISCUSSION TITLE
Should pod definition replace "dummy_image" when setting up KubernetesExecutor

### DISCUSSION THREAD

**Discussion by ketozhang** [apache/airflow#30230] [Q&A]
The docs states 

> Base image
> 
> A pod_template_file must have a container named base at the spec.containers[0] position, and its image must be specified.

and the sets the base container image to be `dummy_image`. Silly question, are we suppose to leave that as-is? If not, what are example images we can use and what are its requirements?

Attempts
- Left as-is, K8 attempted to pull "dummy_image" which doesn't exist.
- Replace pod image with `debian:bullseye`, getting errors that `airflow` doesn't exist
- In `airflow.cfg` `[kubernetes_exector]`, set `worker_container_repository` and `worker_container_tag` to debian, k8 completely ignores this and use whatever is in pod definition.
- Replace pod image with `apache/airflow:latest`, working... ~~but complains psycopg2 doesn't exist. My DAG is just a single `BashOperator`. ~~  Nevermind I was using slim.

Clearly there are requirements to the image and is unclear in the docs.

**Comment by potiuk** [comment]
Well. You want to run some command - your image should be able to run it. Just a usual container question - you always have to specify what image you run it on. For example if you want to print "Hello world" in your job - you can use docker standard hello-world image. Other "requirements' of the image depend on what you want to do as part of your job.

UPDATE: I saw just the first version of your question (much less verbose) - see below.

**Reply by potiuk** [reply]
Also to add - yes if you want to use airflow capabilities (airflow should be installed in the image in the same version) you run it on so starting for Apache/airflow image is a good idea. Regarding the psycopg2 not existing i guess you're getting some configuration issue. Psycopg2 is definitely part of the airflow image, so I am not sure where you get it from, but if you can reproduce it - opening an issue would be great.

**Reply by potiuk** [reply]
There are some intricacies of k8s configuration that make it a bit convoluted (i.e. how worker_container_tag relate to other parts, there  were some changes in recent version of airflow about the configuration of k8s executor and k8s pod operator (see https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html) - and I am not an expert in that part maybe others can help :).

Also there are some articles you might want to look at for example yesterday - https://medium.com/apache-airflow/passing-data-between-tasks-with-the-kubernetespodoperator-in-apache-airflow-7ae9e3e6675c  and probably you can easily find others where you can find examples of how to configure and run k8s executor.

I think our docs are not perfect - because they are created in voluntary time - mostly by people like you, who struggled with something and updated it - and any improvement there, if things are unclar are most welcome (and if you find out some good explanations - bringing them in form of PR is cool.

**Reply by ketozhang** [reply]
Much appreciated! I'd love to make a PR on this doc page once I gather enough info. Your answer is clear these considerations should be in the docs.

I'd like to recommend a base image to the docs but I don't know the minimum requirements. I don't think a hello-world image would work because I'm unable to use images that doesn't contain airflow. With a default airflow.cfg, you'd also need the meta DB libraries (SQLite; and for me  psycopg2).

The article you included is about KubernetesPodOperator, which is different and I found many uses to be redundant if you're using KubernetesExecutor. If I am understanding correctly the scheduler request the base image pod and that pod request another pod defined in KPO.

**Reply by potiuk** [reply]
Correct. I believe the only requirement is Airflow when you use "airflow" features (but also pod_template is configurable and you migh want to have more requirements). Starting with Airlfow is a good beginning. And if you think that what you discover along the way to put as requirements is a good idea for contributing to the docs.
